google-cloud-storage/OVERVIEW.md

```diff
@@ -563,6 +563,36 @@ require "google/cloud/storage"
 storage = Google::Cloud::Storage.new retries: 10, timeout: 120
 ```
 
+The library by default retries all API requests which are always idempotent on a
+"transient" error.
+
+For API requests which are idempotent only if the some conditions are satisfied
+(For ex. a file has the same "generation"), the library retries only if the
+condition is specified.
+
+Rather than using this default behaviour, you may choose to disable the retries
```

Comment:
Also include information about how users can enable retries on non-idempotent operations. (this is possible right?)

Reply:
We'll be adding it in a separate PR which covers retry configurations for non-idempotent operations overall. I didn't want to club that here because Sandeep has already worked on it.